src/react-components/debug-panel/RtcDebugPanel.js

```diff
@@ -0,0 +1,699 @@
+          title={`stats`}
+        />
+      )}
+    </DataPanel>
+  );
+}
+
+TrackStatsPanel.propTypes = {
+  title: PropTypes.string,
+  data: PropTypes.object,
+  xAxis: PropTypes.string,
+  yAxis: PropTypes.string,
+  stats: PropTypes.object
+};
+
+function TransportPanel({ title, data, candidates, producers, consumers, isButtonEnabled, onRestart }) {
+  return (
+    <CollapsiblePanel key={title} title={title} border url={`${MEDIASOUP_DOC_BASE_URL}#Transport`} data={data}>
+      <PanelMessageButton
+        id={"rtcDebugPanel.restartIceButton"}
+        onClick={onRestart}
+        disabled={!isButtonEnabled}
+        primary
+      />
+      {candidates?.length > 0 && (
+        <CollapsiblePanel key={`Candidates`} title={`Candidates`} row url={`${MDN_DOC_BASE_URL}RTCIceCandidate`} border>
+          {candidates}
+        </CollapsiblePanel>
+      )}
+      {producers?.length > 0 && (
+        <CollapsiblePanel
+          key={`Local producers`}
+          title={`Local producers`}
+          row
+          url={`${MEDIASOUP_DOC_BASE_URL}#Producer`}
+          border
+        >
+          {producers}
+        </CollapsiblePanel>
+      )}
+      {consumers}
+    </CollapsiblePanel>
+  );
+}
+
+TransportPanel.propTypes = {
+  candidates: PropTypes.array,
+  producers: PropTypes.array,
+  consumers: PropTypes.array,
+  data: PropTypes.object,
+  onRestart: PropTypes.func,
+  title: PropTypes.string,
+  isButtonEnabled: PropTypes.bool
+};
+
+function SignalingPanel({ data, onConnect, onDisconnect }) {
+  return (
+    <CollapsiblePanel title={"Signaling"} border data={data}>
+      <PanelMessageButton
+        id={data.connected ? "rtcDebugPanel.disconnectSignalingButton" : "rtcDebugPanel.connectSignalingButton"}
+        onClick={() => {
+          if (data.connected) {
+            onDisconnect();
+          } else {
+            onConnect();
+          }
+        }}
+        disabled={false}
+        primary
+      />
+    </CollapsiblePanel>
+  );
+}
+
+SignalingPanel.propTypes = {
+  data: PropTypes.object,
+  onConnect: PropTypes.func,
+  onDisconnect: PropTypes.func
+};
+
+export default class RtcDebugPanel extends Component {
+  static propTypes = {
+    avatarId: PropTypes.string,
+    store: PropTypes.object,
+    scene: PropTypes.object,
+    intl: PropTypes.object,
+    presences: PropTypes.object,
+    history: PropTypes.object,
+    sessionId: PropTypes.string
+  };
+
+  constructor() {
+    super();
+
+    this.state = {};
+  }
+
+  componentDidMount() {
+    this.props.scene.addEventListener("rtc_event", this.logEvent);
+    this.props.store.addEventListener("statechanged", this.storeUpdated);
+    this.runStats(true);
+  }
+
+  componentWillUnmount() {
+    this.props.scene.removeEventListener("rtc_event", this.logEvent);
+    this.props.store.removeEventListener("statechanged", this.storeUpdated);
+    this.runStats(false);
+  }
+
+  logEvent = e => {
+    const currentLog = this.state?.log?.slice(-200) || [];
```

Comment:
With `this.state` values specified, we know what type the data is (so we can handle an empty case) and that `this.state.log` is defined.

We also *expect* the data to have empty cases on the initial load because the data hasn't been fetched yet.

Reply:
I guess initializing will save the overload of doing this check every time a log event is received. Updated.